src/components/z-offcanvas/index.tsx

```diff
@@ -0,0 +1,63 @@
+import { Component, Prop, h, Event, EventEmitter, Watch, Element } from "@stencil/core";
+
+/**
+ * @slot canvasContent - set the content of the canvas
+ */
+@Component({
+  tag: "z-offcanvas",
+  styleUrl: "styles.css",
+  shadow: true,
+})
+
+export class ZOffcanvas
+{
+  @Element() hostElement: HTMLElement;
+
+  /** display component as overlay. Default: false */
+  @Prop({ reflect: true }) overlay = false;
```

Comment:
Anche qui non metterei un booleano ma farei una prop che permette di capire che il componente può andare in overlay oppure affiancarsi e spostare il contenuto della pagina. In questi casi potete crearvi una enum e usarla per tipizzare la prop